file/writable_file_writer.cc

```diff
@@ -603,6 +615,16 @@ IOStatus WritableFileWriter::WriteBufferedWithChecksum(const char* data,
     }
 #endif
     if (!s.ok()) {
+      // If writable_file_->Append() failed, then the data may or may not
+      // exist in the underlying memory buffer, OS page cache, remote file
+      // system's buffer, etc. If WritableFileWriter keeps the data in
+      // buf_, then a future Close() or write retry may send the data to
+      // the underlying file again. If the data does exist in the
+      // underlying buffer and gets written to the file eventually despite
+      // returning error, the file may end up with two duplicate pieces of
+      // data. Therefore, clear the buf_ at the WritableFileWriter layer
+      // and let caller determine error handling.
+      buf_.Size(0);
```

Comment:
Also set ```buffered_data_crc32c_checksum_``` to 0.

Reply:
Nice catch, buffer and the checksum is paired.